src/test/java/StringManipulationTest.java

```diff
@@ -50,100 +59,152 @@ public void testRemoveNthCharacter1() {
     @Test
     public void testRemoveNthCharacter2() {
         manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
-        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(3, true));
+        assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?",
+                manipulatedstring.removeNthCharacter(3, true));
     }
 
+    // like the tests above, but we test it with a different value for n.
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+        assertEquals("I'd 3tt3 puts0med161s inthis5tr16, rght?",
+                manipulatedstring.removeNthCharacter(5, false));
     }
 
+    // we test for a smaller n.
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("let's try a smaller string");
+        assertEquals("ltstyasalrsrn",
+                manipulatedstring.removeNthCharacter(2, false));
     }
 
+    // we test for a smaller n while maintaning the spacing in the string
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("let's try a smaller string");
+        assertEquals("l t s t y a s a l r s r n ",
+                manipulatedstring.removeNthCharacter(2, true));
     }
 
+    // test to make sure it throws a IndexOutOfBoundsException when n > string
+    // length
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This Test should throw IndexOutOfBoundsException for n > string length");
+        assertThrows(IndexOutOfBoundsException.class, () -> manipulatedstring.removeNthCharacter(100, false));
     }
 
+    // test to make sure it throws a IllegalArgumentException when n < 0
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This Test should throw IllegalArgumentException for n < 0");
+        assertThrows(IllegalArgumentException.class, () -> manipulatedstring.removeNthCharacter(-1, false));
     }
 
     @Test
     public void testGeSubStrings1() {
         manipulatedstring.setString("This is my string");
-        String [] sStings = manipulatedstring.getSubStrings(3, 4);
+        String[] sStings = manipulatedstring.getSubStrings(3, 4);
 
         assertEquals(sStings[0], "my");
         assertEquals(sStings[1], "string");
     }
 
+    // test to make sure that it can scale to more substrings
```

Comment:
Recommend modifying this test case so that it calls getSubStrings(1,6) (i.e. get all the words in a string).